src/components/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,40 @@
+import React = require('react');
+
+export interface CheckBoxProps {
+    text?: string;
+    value?: boolean;
+    boxIcon?: string;
+    tickIcon?: string;
+    onChange?: (value: boolean) => any;
+}
+
+export const CheckBox: React.SFC<CheckBoxProps> = (props) => (
+    <div data-automation-id="CHECKBOX_ROOT"
+         onClick={(event) => executeClickHandler(props.onChange!, !props.value)}>
+        <img data-automation-id="CHECKBOX_BOX"
```

Comment:
Also, I'd prefer `React.StatelessComponent` instead of `React.SFC`. It's synonymous, yet more readable.